include/jsoncons/json.hpp

```diff
@@ -759,210 +839,93 @@ class basic_json
             size_t length = char_traits_type::length(s);
             if (length <= small_string_data::max_length)
             {
-                new(reinterpret_cast<void*>(&data_))small_string_data(s, static_cast<uint8_t>(length));
+                construct_var<small_string_data>(s, static_cast<uint8_t>(length));
             }
             else
             {
-                new(reinterpret_cast<void*>(&data_))string_data(s, length, alloc);
+                construct_var<string_data>(s, length, alloc);
             }
         }
 
         variant(const char_type* s, size_t length, const Allocator& alloc)
         {
             if (length <= small_string_data::max_length)
             {
-                new(reinterpret_cast<void*>(&data_))small_string_data(s, static_cast<uint8_t>(length));
+                construct_var<small_string_data>(s, static_cast<uint8_t>(length));
             }
             else
             {
-                new(reinterpret_cast<void*>(&data_))string_data(s, length, alloc);
+                construct_var<string_data>(s, length, alloc);
             }
         }
         variant(const object& val)
         {
-            new(reinterpret_cast<void*>(&data_))object_data(val);
+            construct_var<object_data>(val);
         }
         variant(const object& val, const Allocator& alloc)
         {
-            new(reinterpret_cast<void*>(&data_))object_data(val, alloc);
+            construct_var<object_data>(val, alloc);
         }
         variant(const array& val)
         {
-            new(reinterpret_cast<void*>(&data_))array_data(val);
+            construct_var<array_data>(val);
         }
         variant(const array& val, const Allocator& alloc)
         {
-            new(reinterpret_cast<void*>(&data_))array_data(val,alloc);
-        }
-        template<class InputIterator>
-        variant(InputIterator first, InputIterator last, const Allocator& a)
```

Comment:
I removed this template function because it is not used and uses a constructor in array_data that does not exist.

Reply:
Right, I've removed this function from master